code/modules/events/disease_outbreak.dm

```diff
@@ -307,25 +324,27 @@
 		stage_prob = max(0.5 * properties["stage_rate"], 1)
 		SetSeverity(properties["severity"])
 		visibility_flags |= HIDDEN_SCANNER
+
 		switch(severity)
 			if("Dangerous", "BIOHAZARD")
 				visibility_flags &= ~HIDDEN_SCANNER
+				SetSpread(DISEASE_SPREAD_CONTACT_SKIN)
 
-		GenerateCure(properties)
-
-		if(transmissibility < ADV_SPREAD_LOW)
-			SetSpread(DISEASE_SPREAD_CONTACT_FLUIDS)
+			else if(transmissibility < ADV_SPREAD_LOW)
+				SetSpread(DISEASE_SPREAD_CONTACT_FLUIDS)
 
-		else if(transmissibility < ADV_SPREAD_MID)
-			SetSpread(DISEASE_SPREAD_CONTACT_SKIN)
+			else if(transmissibility < ADV_SPREAD_MID)
+				SetSpread(DISEASE_SPREAD_CONTACT_SKIN)
 
-		else
-			SetSpread(DISEASE_SPREAD_AIRBORNE)
-			visibility_flags &= ~HIDDEN_SCANNER
+			else
+				SetSpread(DISEASE_SPREAD_AIRBORNE)
+				visibility_flags &= ~HIDDEN_SCANNER
```

Comment:
This doesn't make sense to be embedded into the switch statement like this IMO, just have the switch statement call `GenerateCure()` and return if we match `severity` rather than have all of these new variables be introduced into the switch statement.

Reply:
This is setting the transmissibility (spread) not the severity, based on the value the rand generated in the var.